isovar/genetic_code.py

```diff
@@ -0,0 +1,165 @@
+
+class GeneticCode(object):
+    def __init__(self, name, start_codons, stop_codons, codon_table):
+        self.name = name
+        self.start_codons = set(start_codons)
+        self.stop_codons = set(stop_codons)
+        self.codon_table = dict(codon_table)
+        self._check_codons()
+
+    def _check_codons(self):
+        """
+        If codon table is missing stop codons, then add them.
+        """
+        for stop_codon in self.stop_codons:
+            if stop_codon in self.codon_table:
+                if self.codon_table[stop_codon] != "*":
+                    raise ValueError(
+                        "Stop codon '%s' must translate to '*'" % (
+                            stop_codon,))
+            else:
+                self.codon_table[stop_codon] = "*"
+        for start_codon in self.start_codons:
+            if start_codon not in self.codon_table:
+                raise ValueError(
```

Comment:
Start codons, when they're actually being used to initiate translation, almost always translate as methionine (except for a few leucine exceptions that I don't yet know how to deal with), but within the protein might translate as something else (e.g. CTG = leucine, GTG = valine).